fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractCumulativeLoanScheduleGenerator.java

```diff
@@ -423,6 +427,87 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
                 scheduleParams.getTotalRepaymentExpected().getAmount(), totalOutstanding);
     }
 
+    private void addChargesOnlyRepaymentSchedule(MonetaryCurrency currency, LoanScheduleParams scheduleParams,
+            List<LoanScheduleModelPeriod> periods, Set<LoanCharge> loanCharges, MathContext mc, LoanApplicationTerms loanApplicationTerms,
+            Collection<RecalculationDetail> transactions,
+            LoanRepaymentScheduleTransactionProcessor loanRepaymentScheduleTransactionProcessor, final HolidayDetailDTO holidayDetailDTO,
+            final LocalDate lastRestDate, final LocalDate firstRepaymentDate, final LocalDate idealDisbursementDate) {
+        if (periods.size() > 0) {
+            LoanScheduleModelPeriod lastInstallment = periods.get(periods.size() - 1);
+            // get end date for the schedule from loan charges
+            LocalDate scheduleDueDate = null;
+            if (!(BigDecimal.ZERO.compareTo(loanApplicationTerms.getAnnualNominalInterestRate()) < 0)) {
```

Comment:
Can you please simplify this condition?
Negating whether annual nominal interest rate is higher than 0 is not easy to read. 

Also interest rate cannot be negative.

Reply:
However i am not fully sure why does it matter whether there is interest or not...

Reply:
Hi Adam, here i tried to mimic the condition we have for add charge scenario in LoanChargeWritePlatformService->addCharge, where for n+1 case we are checking whether the loan is interest bearing or not.